.github/workflows/main.yml

```diff
@@ -47,24 +60,86 @@ jobs:
         with:
           extra_nix_config: 'access-tokens = github.com=${{ github.token }}'
 
+      - name: Set version 🏷️
+        run: |
+          echo ${{ needs.prepare.outputs.version }} > ./VERSION
+          echo $(git rev-parse HEAD) >> ./VERSION
+
       - name: Build installer 🛠️
         run: |
-          nix build '.#nixosConfigurations.mysystem.config.system.build.installer'
+          nix build -vL '.#nixosConfigurations.mysystem.config.system.build.installer'
 
       - name: Upload installer 📤
         uses: actions/upload-artifact@v3
         with:
           name: installer
           path: result/tarball/nixos-wsl-installer.tar.gz
 
+  launcher:
+    name: Build Launcher 🛠️
+    needs:
+      - prepare
+      - build
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Download installer
+        uses: actions/download-artifact@v3
+        with:
+          name: installer
+
+      - name: Setup MSBuild
+        uses: microsoft/setup-msbuild@v1.1
+        with:
+          vs-version: '[17.2,'
+
+      - name: Set Versions
+        env:
+          NIXOS_WSL_VERSION: ${{ needs.prepare.outputs.version }}
+        run: |
+          ((Get-Content -path .\Launcher\Launcher\Launcher.csproj -Raw) -replace '1.3.3.7',$env:NIXOS_WSL_VERSION) | Set-Content -Path .\Launcher\Launcher\Launcher.csproj
+          ((Get-Content -path .\Launcher\Launcher-Appx\Package.appxmanifest -Raw) -replace '1.3.3.7',$env:NIXOS_WSL_VERSION) | Set-Content -Path .\Launcher\Launcher-Appx\Package.appxmanifest
+      - name: Build Launcher
+        run: cd Launcher; dotnet publish --self-contained -r win-x64 -c Release -p:ContinuousIntegrationBuild=true -p:Deterministic=true --output Launcher\pkg Launcher
+
+      - name: Bundle installer with Launcher
+        run: |
+          Copy-Item nixos-wsl-installer.tar.gz Launcher\Launcher\pkg\
+          Copy-Item nixos-wsl-installer.tar.gz Launcher\Launcher-Appx\
+      - name: Upload Launcher
+        uses: actions/upload-artifact@v3
+        with:
+          name: launcher
+          path: Launcher/Launcher/pkg
+
+      - name: Build AppX package
+        run: cd Launcher; msbuild Launcher-Appx\Launcher-Appx.wapproj /p:Configuration=Release /p:Platform=x64 /p:ContinuousIntegrationBuild=true /p:Deterministic=true
+
+      - name: Sign package
+        env:
+          APPX_SIGNING_CERT: ${{ secrets.APPX_SIGNING_CERT }}
```

Comment:
Can we store that in sops or similar so that we can retrieve it again? secrets are write only in the settings.

Reply:
I can't even edit the repo settings (or commit to it through git for that matter), so having this updateable _somewhere_ else is probaly a good idea, yes